modules/apps/foundation/frontend-theme/responsive-boat-theme/src/js/main.js

```diff
@@ -0,0 +1,39 @@
+
+// This function gets loaded when everything, including the portlets, is on the page
+
+Liferay.on('allPortletsReady', myPageJS);
+
+// Function Containing All of My Event Handlers
+
+function myPageJS() {
+	let nav = AUI.$('#navigation');
+	let navBtnDiv = AUI.$('#navigation div.nav-btn');
+	let portletParagraphs = AUI.$('#content .portlet-body p');
+
+	document.onscroll = function() {
+		// Toggle 'scrolled' class on nav based on scrolled location
+		if (window.scrollY >= 140 && !nav.hasClass('scrolled'))
+			nav.addClass('scrolled');
+		else if (window.scrollY < 140 && nav.hasClass('scrolled'))
+			nav.removeClass('scrolled');
+
+		// Toggle 'visible' class to make portlet content visible when scrolled to
+		for (let i = 0; i < portletParagraphs.length; i++) {
+			if (!portletParagraphs.eq(i).hasClass('visible') && portletParagraphs[i].getBoundingClientRect().top - window.innerHeight <= 0)
+				portletParagraphs.eq(i).addClass('visible');
+		}
```

Comment:
Using a similar logic, is there anything that we can do to optimize this block of JS?

Reply:
I'm thinking the best way to apply that logic is like so:

```
var hasCssVisible = portletParagraphs.eq(i).hasClass('visible');
var scrollPositionY = portletParagraphs[i].getBoundingClientRect().top;

var isVisible = hasCssVisible || scrollPositionY > window.innerHeight;

portletParagraphs.eq(i).toggleClass('visible', isVisible);
```
<table>
<thead>
<td><strong>Has class</strong></td>
<td><strong>Scroll Y > winHeight</strong></td>
<td><strong>Result</strong></td>
</thead>
<tbody>
<tr><td>false</td><td>false</td><td>false - remove class</td></tr>
<tr><td>false</td><td>true</td><td>true - add class</td></tr>
<tr><td>true</td><td>true</td><td>true - add class</td></tr>
<tr><td>true</td><td>false</td><td>true - add class</td></tr>
</tbody>
</table>

Reply:
We can probably cache ```portletParagraphs.eq(i)```.

Reply:
We should probably update the JS variable to be ```hasCssClassVisible```.